### DISCUSSION TITLE
VFDB gives me a different annotation than BAKTA?

### DISCUSSION THREAD

**Discussion by reymonera** [oschwengers/bakta#188] [Help wanted]
Hi,

First of all, thanks for the tool. BAKTA is awesome.

So, I'm opening this thread (hopefully in the correct place) because I've been using the VFAnalyzer in the VFDB database to detect virulence genes. However, when I tried to find these genes through my .faa file from BAKTA, the featured names are certainly different, and even there are some extra aa in the sequence.

From what I could understand, VFAnalyzer seems to use a taxonomic approach towards its annotation. So, for example, if I give it a certain genus, VFAnalyzer will have a DB for this genus and will try to use it to detect these genes. 

So far, the annotations seem to be similar since the names seem to describe something similar, but it certainly changes things if VFanalyzer gives a certain aa sequence a name instead of the name featured in my annotation. To top it all, whenever I blast-p my sequences, I obtain another name.

So I just wanted to know which DB should I trust or maybe I am doing something badly. I'm kind of a novice user, so any help will be welcomed.

Thanks!

Camila

**Comment by oschwengers** [comment]
Hi Camilla,
thanks a lot for asking. Unfortunately, your description is a little bit vague, so I'm not quite sure if I fully understand. Could you maybe provide an example with detailed information on which protein products Bakta provides and which VFAnalyzer provides?
Indeed, we incorporated the core set of VFDB, i.e. the experimentally verified ones, into the expert protein system. Hence, in principle Bakta should provide the same product descriptions. But as you already mentioned, VFDB also has genus specific databases and thus, it could be that product descriptions slightly differ.

Could you provide some examples, so that I could take deeper look at this?
Thanks again!
Oliver

**Comment by reymonera** [comment]
Hi!

Sorry for returning late to this discussion. Yes, of course, I can provide an example. As a brief introduction, I'm trying to identify virulent genes in the genome of Bartonella bacilliformis. VFAnalyzer gives me a .faa file along with a table and identified orfs with their respective names. For example, for a genome, VFAnalyzer gave me an orf and annotated it as the "ialB" gene.

However, when I decided to search for said gene in my BAKTA annotations, it appears as "bamA". Apparently, both genes are related but it still leaves me uncertain on which label I should use.

It also appears that VFAnalyzer has classified a bunch of genes as "ialB", but in BAKTA, there are 2 descriptions that fit in this label: "ialB" and "Invasion associated locus B (IalB) protein". So the group of genes identified in the VFAnalyzer as a single group are 2 groups in BAKTA. I have aligned both groups and they are significally different, although still related.

I'm not sure with which results I should continue, since apparently BAKTA also uses the VFDB and I'm quite curious on why they haven't shown the same results.

**Reply by oschwengers** [reply]
Hi, for a detailed answer I'd need to take a deeper look at the Bakta results of these two genes. Could you provide the `*.json` file or at least the two `feature` sections of the JSON file of these genes?

It always depends on the annotation quality of the underlying databases and w/o any further details, I can only speculate: In this special case, relevant DBs might be UniRef100, UniRef90, NCBIfams, NCBI AMR genes and VFDB. So it might be the case that one DB has more detailed annotations for these two related but distinct genes than other DBs. It's hard to judge which one is the better.

**Reply by reymonera** [reply]
Hi Oliver,

Of course. I'll copy the files of my example here.

So, in my vfdb analysis, I obtained the following description in a tabular file:
<google-sheets-html-origin><style type="text/css"></style>

orf_vfdb | related_genes | vfdb_class | vfdb_factor | bakta_num | bakta | bakta_des
-- | -- | -- | -- | -- | -- | --
orf00668 | ialB | Adherence | Invasion-associated locus B | KBBJEN_03195 | bamA | outer membrane protein assembly factor BamA

This orf00668 appears in the .faa file obtained from the VFAnalyzer like this:

```
>orf00668	draft|bb25|Bartonella|orf00668	NZ_KK097689.1
MYVQGIKYKVKKPMATNSKFFNAASMLVLAKVTAVPIIAAIPVSIIGNAQASVVQSIEIH
GNSLVDAKTIQDNISIKAGKTFESSDINAAVKQLFGLGLFYDVKINQVDNKLIVTVKEYE
IINQVLFQGNRALKEPDLKRFITLKPNTTFSAAKLSADIKTIKDAYRTVGRSNISVSAQT
INLGKGRVNVIFTIVEGQRTKIKNIVFEGNKAFGNRRLHDVISTKSSGIFSLLMRGDVYT
EERLAADEQALRLFYYNRGYADFRVVSSKALFDEISNSYEIYFVLDEGARYKIGNVGVES
DIDGIDTKSMTNMLKTRPGDVYSAERIERSVAIINDKVADSGYAFAKVDPRGNRDFQNHT
ISILYNIEQGPRVYIQRIDIRGNEKTRDNVIRREIDLNEGDAYNQTLVRRAQRRLESLGF
FKAVNISMVPTSESDQVTLVIDVVETPTGDFSFSGGYTTGGNSPGMSLEASVTERNLGGY
GQYARLGLGAGQGNSRNYNFSFVEPYFLGYRLSSGVDIFRSTSRVDKAYDSQQTGGSLRF
GVSINEEISASLAYSYVQEEYNFGSNYDLTDDAVIEELYDKYSGAIVQAAKNSPWRRSSV
IFGLTYNSVDNMRNPHDGLYIHALQEYAGLGGNAKFLKTTGKAIMYKTLSDEMDLVGLLS
IGAGYMHEIGKDGVRIFDMFKSNTDIIRGFRYNGIGPRQISNTGEVSFLGGTTYMNATAE
AQFPIPLIPDSLGFRGAVFADAATLYGNNYKPVLQNEKPVTDRGSAWRTSAGVSLMWESP
FGPLRVDYAWPITKQNGDQVQNLNFGISTKF
```

When I searched for it in my BAKTA .faa file, it matched with this sequence:

```
>KBBJEN_03195 outer membrane protein assembly factor BamA
MYVQGIKYKVKKPMATNSKFFNAASMLVLAKVTAVPIIAAIPVSIIGNAQASVVQSIEIHGNSLVDAKTIQDNISIKAGKTFESSDINAAVKQLFGLGLFYDVKINQVDNKLIVTVKEYEIINQVLFQGNRALKEPDLKRFITLKPNTTFSAAKLSADIKTIKDAYRTVGRSNISVSAQTINLGKGRVNVIFTIVEGQRTKIKNIVFEGNKAFGNRRLHDVISTKSSGIFSLLMRGDVYTEERLAADEQALRLFYYNRGYADFRVVSSKALFDEISNSYEIYFVLDEGARYKIGNVGVESDIDGIDTKSMTNMLKTRPGDVYSAERIERSVAIINDKVADSGYAFAKVDPRGNRDFQNHTISILYNIEQGPRVYIQRIDIRGNEKTRDNVIRREIDLNEGDAYNQTLVRRAQRRLESLGFFKAVNISMVPTSESDQVTLVIDVVETPTGDFSFSGGYTTGGNSPGMSLEASVTERNLGGYGQYARLGLGAGQGNSRNYNFSFVEPYFLGYRLSSGVDIFRSTSRVDKAYDSQQTGGSLRFGVSINEEISASLAYSYVQEEYNFGSNYDLTDDAVIEELYDKYSGAIVQAAKNSPWRRSSVIFGLTYNSVDNMRNPHDGLYIHALQEYAGLGGNAKFLKTTGKAIMYKTLSDEMDLVGLLSIGAGYMHEIGKDGVRIFDMFKSNTDIIRGFRYNGIGPRQISNTGEVSFLGGTTYMNATAEAQFPIPLIPDSLGFRGAVFADAATLYGNNYKPVLQNEKPVTDRGSAWRTSAGVSLMWESPFGPLRVDYAWPITKQNGDQVQNLNFGISTKF
```
But, as you can see, it is not labeled as ialB, but as bamA. However, in the whole .faa file made in BAKTA, the ialB gene is, indeed, present.

Here's the .json file of the annotation:

[BAKTA_bb25fna.json.zip](https://github.com/oschwengers/bakta/files/10875889/BAKTA_bb25fna.json.zip)

Thanks for the help,

Regards,

Camila

**Reply by oschwengers** [reply]
In this case, VFDB just has a different description than UniProt. Gene symbols like `bamA` / `ialB` can be different in different species though they describe the same protein. Often this has mere historic reasons. But, unfortunately, in this case also the product description differ. Since I'm not an expert in this species, I cannot tell which one is the correct or better one.

Often, UniProt has a bunch of information along with the sequences. Hence, I'd refer you to the seed&representative member of the UniRef90 cluster of your hit: https://www.uniprot.org/uniprotkb/A0A072R4F0/entry. There you get tons of additional information and further links to other databases, especially InterPro and Pfam providing protein domains.
